### ISSUE TITLE
Feature: user email notification upon app password creation

### ISSUE THREAD

**Issue by rriemann** [nextcloud/server#44769]
**Is your feature request related to a problem? Please describe.**

A person leaves an online session unattended and an attacker creates secretly an app password. Later, the attacker can download most data of the user at any time from any place.

**Describe the solution you'd like**
10 minutes after the app password creation (so that the attacker does not see the notification to delete it immediately), the user receives an email notification about the app password creation with a proposition to review/revoke the app passwords in case they have not created the password themselves.

**Describe alternatives you've considered**

I have not considered alternatives. For those who do not require app passwords, it would be also nice to disable user app password creation altogether. I believe this alternative was already brought up in another issue, but cannot find it currently.

**Comment by joshtrichards** [comment]
> A person leaves an online session unattended and an attacker creates secretly an app password.

I get prompted for my password before I'm allowed to create a new app password. Are you not being prompted?

**Comment by rriemann** [comment]
> > A person leaves an online session unattended and an attacker creates secretly an app password.
> 
> I get prompted for my password before I'm allowed to create a new app password. Are you not being prompted?

No, it doesn't prompt for a password. With SSO, users don't have passwords.

**Comment by joshtrichards** [comment]
Related: #34444
